content/en/docs/tutorials/microservices-3-templating-and-multi-env.md

```diff
@@ -0,0 +1,359 @@
+metadata:
+  name: adservice
+  namespace: ms-demo-{{ target.name }}
+...
+```
+
+This can however easily lead to resources being forgotten or resources where you have no control over, e.g. rendered
+Helm Charts. Another way to set the namespace on multiple resources is by using the
+[`namespace` property](https://kubectl.docs.kubernetes.io/guides/config_management/namespaces_names/) of kustomize.
+For example, instead of changing the `adservice` deployment directly, you could for example modify the content of
+`services/adservice/kustomization.yml` to:
+
+```yaml
+resources:
+  - deployment.yml
+  - service.yml
+
+namespace: ms-demo-{{ target.name }}
+```
+
+This is better than the naive solution, but still limited in a comparable (but not as bad) way. The most powerful and
+preferred solution is use [`overrideNamespace`]({{< ref "docs/reference/deployments/deployment-yml#overridenamespace" >}})
+in the root `deployment.yml`:
+
+```yaml
+...
+overrideNamespace: ms-demo-{{ target.name }}
+...
+```
+
+As an alternative, you could also use `overrideNamespace` separately in `third-party/deployment.yml` and
+`services/deploymetn.yml`. In this case, you're also free to use different prefixed for the namespaces, as long as you
+include `{{ target.name }}` in them.
+
+{{< alert >}}
+Please note that `overrideNamespace` only takes effect on a kustomize deployment if it does NOT specify `namespace`.
+If you followed the `kustomization.yml` example from above, make sure to undo the changes to `kustomization.yml`.
+{{< /alert >}}
+
+## Helm Charts and namespaces
+The previously described way of making namespaces dynamic in all resources works well for most cases. There are however
+situations where this is not enough, mostly when the name of the namespace is used in other places then `metadata.namespace`.
+
+Helm Charts very often do this internally, which makes it necessary to also include the dynamic namespace into the
+`helm-chart.yml`'s `namespace` property. You will have to do this for the redis chart as well, so let's modify
+`third-party/redis/helm-chart.yml` to:
+
+```yaml
+helmChart:
+  repo: https://charts.bitnami.com/bitnami
+  chartName: redis
+  chartVersion: 16.8.2
+  releaseName: cart
+  namespace: ms-demo-{{ target.name }}
+  output: deploy.yml
+```
+
+Without this change, redis is going to be deployed successfully but will then fail to start due to wrong references
+to the `default` namespace internally.
+
+## Making commonLabels unique per target
+[`commonLabels`]({{< ref "docs/reference/deployments/deployment-yml#commonlabels" >}}) in your root `deployment.yml` has
+a very special meaning which is important to understand and work with. The combination of all `commonLabels` MUST be unique
+between all supported targets on a cluster, including the ones that don't exist yet and are from other kluctl projects.
+
+This is because kluctl uses these to identify which resources on a cluster belong to the currently processed deployment,
```

Comment:
repetitive use of "which". Just use "what resources"

Reply:
changed it to: This is because kluctl uses these to identify resources belonging to the currently processed deployment/target, which becomes especially important when deleting or pruning.